pkg/jx/cmd/delete_extension.go

```diff
@@ -144,40 +146,56 @@ func (o *DeleteExtensionOptions) Run() error {
 	if len(args) == 0 {
 		return fmt.Errorf("Specify the extensions to delete using %s or delete them all using %s.", util.ColorInfo("jx delete extensions <extension name>"), util.ColorInfo("jx delete extension --all"))
 	}
-	deleteExtensions := strings.Join(args, ", ")
 
-	if !o.BatchMode && !util.Confirm(fmt.Sprintf("You are about to delete the Extensions: %s", deleteExtensions), false, "The list of Extensions to be deleted", o.In, o.Out, o.Err) {
-		return nil
+	configLookup := make(map[string]jenkinsv1.ExtensionConfig)
+	for _, c := range extensionsList.Extensions {
+		configLookup[c.Name] = c
 	}
+	extensionsToDelete := make(map[string]*jenkinsv1.Extension)
 	for _, name := range args {
 		if util.StringArrayIndex(names, name) < 0 {
 			return util.InvalidOption(optionLabel, name, names)
 		}
 		ext, _ := lookup[name]
-		if ext.IsOnUninstall() {
+		err := extensions.GetAndDeduplicateChildrenRecursively(ext, lookupByUUID, extensionsToDelete)
+		if err != nil {
+			return err
+		}
+	}
+
+	extensionsToDeleteStrings := make([]string, 0)
+	for _, e := range extensionsToDelete {
+		extensionsToDeleteStrings = append(extensionsToDeleteStrings, e.Spec.FullyQualifiedName())
+	}
+
+	// TODO display the extensions to delete in a tree view
+	if !o.BatchMode && !util.Confirm(fmt.Sprintf("You are about to delete the Extensions: %s",
```

Comment:
Nit: Autoformatting isn't always as good as humans. Try to keep different methods/parameter-sets on the same line, eg:
```
if !o.BatchMode && !util.Confirm(
        fmt.Sprintf("You are about to delete the Extensions: %s", strings.Join(extensionsToDeleteStrings, ", ")), 
        false, list of Extensions to be deleted", o.In, o.Out, o.Err) {
```
or some variant thereof reads easier to me

Reply:
I hate multi-line statements, so tbh it's greek to me ;-)